src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs

```diff
@@ -102,9 +113,11 @@ public void AddDirectory (string folder, string folderInArchive, CompressionMeth
 				var internalDir = dir.Replace (folder, string.Empty);
 				string fullDirPath = folderInArchive + internalDir;
 				try {
-					zip.CreateDirectory (fullDirPath);
+					string dirPathInZip = TrimEntryName (fullDirPath, appendSeparator: true);
+					if (CreateDirectoriesInZip && !zip.ContainsEntry (dirPathInZip))
+						zip.CreateDirectory (dirPathInZip);
 				} catch (ZipException) {
-					
+
```

Comment:
Shouldn't we *log* the exception?  Silent errors are a *menace*. 

(And *yes*, this should have been noted in 86888322a46261193d4006a97aedd34acee4f921.  My bad.)